### ISSUE TITLE
Where update?

### ISSUE THREAD

**Issue by Toxenskiy** [UnamSanctam/SilentXMRMiner#424]
It's already September 11th.

**Comment by UnamSanctam** [comment]
It's not exactly an update but an entirely new miner, I have to find some way to get what GPUs the computer has in C. The install and injection parts are already complete.

**Comment by Toxenskiy** [comment]
and you do something like that, which I sent with a screenshot a long time ago

**Comment by Toxenskiy** [comment]
![image](https://user-images.githubusercontent.com/55586522/132945285-9af7a5b3-9b5a-416f-ad1d-95ed9f935770.png)

**Comment by Toxenskiy** [comment]
How did you decide to make a website panel or a software panel

**Comment by UnamSanctam** [comment]
It does not get everything in your screenshot no since it's not really necessary, the panel only controls the miner and not the computer since you would just use a RAT for that. The API and panel are already technically done and have been for about a month but I'm working on the new installer/injector now. The panel uses HTTP endpoints so it's a website panel.

**Comment by Toxenskiy** [comment]
You can send a couple of screenshots interesting to see.

**Comment by Toxenskiy** [comment]
?

**Comment by Toxenskiy** [comment]
You add function full update build?

**Comment by UnamSanctam** [comment]
What? I will also repost a status update I wrote to someone else,

>Currently I'm working on getting Icons and Assembly Data working, it currently exists in my Binder and Downloader but the problem is that what it's using (windres) does not support having spaces in the path it's called from, this has been an issue sifor a long time but hasn't been fixed yet. I've sort of circumvented it by using short paths but not all computers support it (either they have turned off short paths or they don't work on that computer). So I'll probably have to code (or find some code for a fix) and compile a custom build of binutils where that issue is fixed But currently the installer, injector, GPU check and running miners checks are already done in my C miner version so it's not a lot left. Though I have to update xmrig to the newest version which is quite a bit of work.

**Comment by Toxenskiy** [comment]
Well, as far as I understand, you did a lot in xmrig, and since this is an open source project, it is worth uploading the source xmrig.

**Comment by Toxenskiy** [comment]
Well, what's up with the update?

**Comment by UnamSanctam** [comment]
I'm working on it, I have to change the builder to build C files now using the binder/download, which is the reason I made those two in the first place, as a test to see how well it would work.

**Comment by Toxenskiy** [comment]
Well, your predictions, how soon will you update?

**Comment by UnamSanctam** [comment]
No idea, I have no clue whether what I'm attempting will work nor how much I have to remake to make it as undetectable as possible. I have to create it, scan it with antiviruses, remake it and scan it, then redo that process over and over again which is maybe 40-80 hours of work usually (to get it to 0 detections, even the strict antiviruses, though that might not be possible due to some antiviruses detecting RunPE).

**Comment by Toxenskiy** [comment]
Well, knowing how much free time you have, it will take at least a month.

**Comment by UnamSanctam** [comment]
Yeah depends, I have the entire weekend free to work on it.

**Comment by Toxenskiy** [comment]
Why did you settle for a .net domain, and not cheaper analogs, or did you buy it a long time ago?

**Comment by Toxenskiy** [comment]
?

**Comment by UnamSanctam** [comment]
Because that's one of the cheapest one from where I bought it.

**Comment by Toxenskiy** [comment]
where did you buy? taurs domen . com?

**Comment by UnamSanctam** [comment]
https://njal.la/

**Comment by Toxenskiy** [comment]
well, such a panel would look much more interesting 

![image psd](https://user-images.githubusercontent.com/55586522/134352889-039f7d71-1324-4758-942b-16f4531e9879.png)

**Comment by UnamSanctam** [comment]
Well that's a RAT with a miner built in, the current panel will not be a RAT.

**Comment by UnamSanctam** [comment]
Though one project planned is a RAT (though probably far in the future since I need to create one and all the features in C from scratch) which probably will also have the miner built into it.

**Comment by Toxenskiy** [comment]
?

**Comment by UnamSanctam** [comment]
Good, I still have to modify a zip library to create a way to do memory to memory decompression.

**Comment by Toxenskiy** [comment]
By the way, have you tried to do it in the eth miner, just check the memory of the video card, that is, if less than 6, then it is mined, etc., If more than 6, then etc.

**Comment by UnamSanctam** [comment]
Well, there is no guarantee that someone wants to mine ETH just because they use the ethash algorithm, they could be mining ETHO or anything else or they could even be mining different coins dynamically with something like zergpool. It's technically possible that I could add it into the unified builder where you could enter the max VRAM for each miner (so you would add two miners, one below a specific VRAM and one above) but the problem is that there isn't really any good way to get the VRAM available. If you try getting it with WMI then it can only show a maximum of 4GB of VRAM even if you have more since it uses 32-bit integers, the only other real way is to check a specific key in the registry with powershell and convert it to a readable GB format, but it does not work on all computers from my tests which means that the miner wouldn't work on all computers.

**Comment by Toxenskiy** [comment]
Did you launch the miner again, only after rebooting the computer?

**Comment by Toxenskiy** [comment]
?

**Comment by Toxenskiy** [comment]
?

**Comment by UnamSanctam** [comment]
What do you mean by?
> Did you launch the miner again, only after rebooting the computer?

**Comment by Toxenskiy** [comment]
well, my miner starts, only after restarting windows

**Comment by Toxenskiy** [comment]
and where is api?

**Comment by UnamSanctam** [comment]
Will be in the unified miner since I need to create tutorials and everything.

**Comment by Toxenskiy** [comment]
and when will it be?

**Comment by Toxenskiy** [comment]
?

**Comment by ssk666** [comment]
> well, such a panel would look much more interesting
> 
> ![image psd](https://user-images.githubusercontent.com/55586522/134352889-039f7d71-1324-4758-942b-16f4531e9879.png)

What's the rat on ur pic?

**Comment by Toxenskiy** [comment]
modded acyncrat

**Comment by ssk666** [comment]
> modded acyncrat

Thank u man

**Comment by Toxenskiy** [comment]
> Will be in the unified miner since I need to create tutorials and everything.

and when will it be?

**Comment by UnamSanctam** [comment]
I can't know since I don't know when everything is done, I've been working on decreasing detections for the last 2 days and I've succeeded a but preferably I'd want to make it pretty much undetectable at release.

**Comment by Toxenskiy** [comment]
well, less than two weeks?

**Comment by Toxenskiy** [comment]
> I can't know since I don't know when everything is done, I've been working on decreasing detections for the last 2 days and I've succeeded a bit but preferably I'd want to make it pretty much undetectable at release.

well, less than two weeks?

**Comment by UnamSanctam** [comment]
I'm just bruteforcing different methods and changes to try and reduce it so I'm not sure. I've finished fixing all the system calls to be actual system calls which brings down detections but I had to spend quite some time coding assembler which drained a lot of my energy. Right now I'm just making changes, compiling it and then testing it over and over again until I can find and change whatever is causing detections.

**Comment by Toxenskiy** [comment]
It is a pity that after the release, the stub will be dead.

**Comment by UnamSanctam** [comment]
Well that's how it is for all programs, there isn't a single program that isn't the case for since antivirus companies is constantly improving their detections. Though currently (the latest two updates) are pretty much undetectable in most cases, especially with "Bypass Windows Defender" enabled. But I'd preferably want it to be able to be "installed" without getting detected at all before I release anything since otherwise I'll just have to work on it and update it constantly.
